estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -510,55 +525,140 @@ def open_binding(
     binding_index: int,
     state: _ResourceState,
     task: Task,
-    fetch_changes: FetchChangesFn[_BaseDocument] | None = None,
-    fetch_page: FetchPageFn[_BaseDocument] | None = None,
-    fetch_snapshot: FetchSnapshotFn[_BaseDocument] | None = None,
+    fetch_changes: FetchChangesFn[_BaseDocument]
+    | dict[str, FetchChangesFn[_BaseDocument]]
+    | None = None,
+    fetch_page: FetchPageFn[_BaseDocument]
+    | dict[str, FetchPageFn[_BaseDocument]]
+    | None = None,
+    fetch_snapshot: FetchSnapshotFn[_BaseDocument]
```

Comment:
I don't think we should handle multiple cursors for a `fetch_snapshot`, at least not right now. These "snapshot" bindings use a `_meta/row_id` key and I'm not sure how we would make that work with multiple snapshot processes writing documents into the same collection. I'm not sure if it really could be made to work.

Reply:
Ah okay. I'll remove this changes related to snapshots. I need to dig more into the quirks so I understand this a bit more.

Reply:
Done.